### PR TITLE
cl: preserve zero values for blank struct fields

### PR DESCRIPTION
**Pull request by cpunion**
## Summary

- emit zero constants for blank fields in static struct initializers
- discard dynamic stores into descendants of blank aggregate fields while preserving function-local right-hand-side side effects
- remove the macOS/arm64 and Linux/amd64 `fixedbugs/issue31546.go` run xfails

## Review counterexamples

The tests directly cover blank fields, nested blank structs, array descendants reached through `IndexAddr`, adjacent non-blank fields, and side-effecting function-local initializers. `IndexAddr` traversal is limited to pointer-to-array storage and has a defensive slice/backing-array boundary counterexample that failed before the review fix. The runtime fixture uses unsafe layout checks instead of reflect so it also builds for embedded targets.

## Validation after rebasing onto latest main

- focused compiler, classifier, and blank-field runtime fixture tests
- focused `isBlankFieldStore` coverage: 94.1%
- `fixedbugs/issue31546.go` without xfail on Go 1.26.5 on macOS/arm64
- Linux/amd64 Go 1.26.5 is covered by the PR matrix
- `git diff --check`

### CODE REVIEW

File: cl/rewrite_internal_test.go [hunk 1]
```diff
@@ -126,6 +126,53 @@ func Use() string {
...
+		"@staticinit.Flat = global %staticinit.Nested { i64 1, i64 0, i64 3 }",
+		"@staticinit.Deep = global %staticinit.Outer { i64 4, %staticinit.Nested zeroinitializer, i64 8 }",
+	} {
+		if !strings.Contains(ir, want) {
+			t.Fatalf("blank field was not zeroed in static initializer %q:\n%s", want, ir)
+		}
+	}
+}
+
+func TestBlankFieldStoreRejectsInvalidAddresses(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cpunion) [reply cl/rewrite_internal_test.go:249]: Added TestBlankFieldStoreClassification in 263a40829. It builds real go/ssa for direct blank fields, nested descendants, array descendants, and a non-blank sibling, then checks isBlankFieldStore directly. It also carries the defensive slice-boundary counterexample from the other thread. Focused function coverage is 94.1%.

File: cl/compile.go [hunk 2]
```diff
@@ -1843,13 +1843,26 @@ func (p *context) compileValue(b llssa.Builder, v ssa.Value) llssa.Expr {
...
+		case *ssa.FieldAddr:
+			_, st, ok := fieldAddrStruct(current)
+			if !ok {
+				return false
+			}
+			if st.Field(current.Field).Name() == "_" {
+				return true
+			}
+			addr = current.X
+		case *ssa.IndexAddr:
```
[the remark this answers is not in the document]
Comment (cpunion) [reply cl/compile.go:1983]: Addressed in 263a40829. IndexAddr now walks upward only when X is a pointer to an array, so it cannot cross a slice header into separately allocated backing storage. The new classification test includes a defensive synthetic slice-boundary case; it failed on the previous implementation and now passes. The containment rule is also documented on isBlankFieldStore.
